src/cursor/cur_stat.c

```diff
@@ -565,6 +565,8 @@ __wt_curstat_init(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *curjoin,
         return (0);
     }
 
+    /* Data source statistics are only available after recovery completes. */
+    WT_ASSERT(session, F_ISSET(S2C(session), WT_CONN_RECOVERY_COMPLETE));
```

Comment:
As we opened a handle_event for MongoDB to query the stats during the recovery and shutdown, it is possible that they may query the data source statistics. Instead of the assert, how about raising an error?

Reply:
Or we can use `WT_ASSERT_ALWAYS`